cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go

```diff
@@ -46,17 +41,7 @@ func main() {
 	flag.Parse()
 	glog.Info("Kubernetes Elasticsearch logging discovery")
 
-	settings, err := clientcmd.LoadFromFile(*kubeconfig)
-	if err != nil {
-		glog.Fatalf("Error loading configuration from %s: %v", *kubeconfig, err.Error())
-	}
-
-	config, err := clientcmd.NewDefaultClientConfig(*settings, &clientcmd.ConfigOverrides{}).ClientConfig()
-	if err != nil {
-		glog.Fatalf("Failed to construct config: %v", err)
-	}
-
-	c, err := client.New(config)
+	c, err := client.NewInCluster()
```

Comment:
Yeah-- in another commit I have changes to kubectl code; people that use the same code path should be covered, but that is a very confusing bit of excessive overriding.

But this bit of code has essentially no usefulness outside of a cluster, so I think it's OK.

@satnam6502 you may want to glance at this.